### PR TITLE
Go project skeleton

### PR DESCRIPTION
**Pull request by skhalash**
# What's done?

Just a simple hello-world-like web service application. Added a Dockerfile a Makefile. Nothing fancy, but should be enough to test a CI pipeline against.

### CODE REVIEW

**Review by cfreemoser — CHANGES_REQUESTED** [review]
Let's merge my branch first into yours

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		GOOS=$(os) GOARCH=amd64 go build -o ./bin/$(BINARY)-$(os)-amd64 ./cmd/injector/main.go
+
+# Runs all of the tests
+.PHONY: test
+test:
+		go test -v ./...
+
+# Builds a docker image
+.PHONY: image
+image:
```
Comment (cfreemoser) [inline Makefile]: maybe rename it to docker

Comment (cfreemoser) [inline Makefile]: adding make lint

File: .github/workflows/dockerpush.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - v*
+
+  # Run tests for any PRs.
+  pull_request:
+
+env:
+  # TODO: Change variable to your image's name.
+  IMAGE_NAME: valtron-injector
+
+jobs:
```
Comment (cfreemoser) [inline .github/workflows/dockerpush.yml:20]: add lint

File: internal/injector/server_test.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	request := httptest.NewRequest(http.MethodPost, "/mutate", strings.NewReader(validAdmissionReview))
+	recorder := httptest.NewRecorder()
+
+	sut.ServerHTTP(recorder, request)
+
+	admitterMock.AssertNumberOfCalls(t, "Admit", 1)
+}
+
+func TestFailIfAdmitterFails(t *testing.T) {
+	admitterMock := &mocks.Admitter{}
```
Comment (cfreemoser) [inline internal/injector/server_test.go]: It is a stub, rename the variable

File: internal/injector/server_test.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	request := httptest.NewRequest(http.MethodPost, "/mutate", strings.NewReader(validAdmissionReview))
+	recorder := httptest.NewRecorder()
+
+	sut.ServerHTTP(recorder, request)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Result().StatusCode)
+}
+
+func TestSucceedIfAdmitterSucceeds(t *testing.T) {
+	admitterMock := &mocks.Admitter{}
```
Comment (cfreemoser) [inline internal/injector/server_test.go]: It is a stub, rename the variable

**Review by cfreemoser — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (skhalash) [reply Makefile]: Yeah, makes sense

Comment (skhalash) [reply Makefile]: Done

File: .github/workflows/dockerpush.yml (hunk 2 above)
Comment (skhalash) [reply .github/workflows/dockerpush.yml:20]: Done

**Pull request merged**